tests/test_otio_openassetio.py

```diff
@@ -162,9 +185,7 @@ def test_when_no_manager_setting_and_no_default_config_then_error_raised(
     linker_args_no_settings, monkeypatch
 ):
     monkeypatch.delenv(ManagerFactory.kDefaultManagerConfigEnvVarName, raising=False)
-    with pytest.raises(
-        RuntimeError, match="No default OpenAssetIO manager configured"
-    ):
+    with pytest.raises(RuntimeError, match="No default OpenAssetIO manager configured"):
```

Comment:
[Question] Whilst were here, should this also be a `ConfigurationError`? Sorry missed this before (or make the other a RuntimeError so it's consistent?)

Reply:
Good spot. Used `ConfigurationError` in 8bd8066e.

There is a broader question of who should use OpenAssetIO exceptions.  I.e. host integration points, middleware like this OTIO plugin, and manager implementations - vs. internal OpenAssetIO functions.  Would be good to come up with some guidance.

Reply:
As a starting point
* Host integration points - depends on how loosely coupled.  If a certain app has its own decoupled asset API, they should use OpenAssetIO exceptions where its an OpenAssetIO-specific issue.
* Middleware (ours) - use OpenAssetIO exceptions
* Middleware (third-party) - context-dependent, but probably same advice as for host integration points
* Managers - use OpenAssetIO exceptions

Reply:
https://github.com/OpenAssetIO/OpenAssetIO/issues/1184